spec/terrafying/components/vpc_spec.rb

```diff
@@ -0,0 +1,217 @@
+    end
+
+    it "should create nat gateway public networks when internet accessible" do
+      vpc = Terrafying::Components::VPC.create("foo", "10.0.0.0/16", { internet_access: true })
+
+      expect(vpc.subnets[:nat_gateway].count).to eq(@azs.count)
+    end
+
+    it "should be able to be overriden by options" do
+      vpc = Terrafying::Components::VPC.create(
+        "foo", "10.0.0.0/16", {
+          subnets: {
+            dmz: { public: true },
+            secure: { public: false, internet: false },
+          },
+        }
+      )
+
+      expect(vpc.subnets.has_key?(:public)).to be false
+      expect(vpc.subnets.has_key?(:private)).to be false
+      expect(vpc.subnets[:dmz].count).to eq(@azs.count)
+      expect(vpc.subnets[:secure].count).to eq(@azs.count)
+    end
+
+  end
+
+  it "should create a security group for SSH around the VPC" do
+    cidr = "10.1.0.0/16"
+    vpc = Terrafying::Components::VPC.create("foo", cidr)
+
+    expect(vpc.output["resource"]["aws_security_group"].count).to eq(1)
+
+    ssh_security_group = vpc.output["resource"]["aws_security_group"].values.first
+
+    expect(ssh_security_group[:ingress].count).to eq(1)
+    expect(ssh_security_group[:egress].count).to eq(1)
+
+    rule = {
+      from_port: 22,
+      to_port: 22,
+      protocol: "tcp",
+      cidr_blocks: [cidr],
+    }
+
+    expect(ssh_security_group[:ingress][0]).to eq(rule)
+    expect(ssh_security_group[:egress][0]).to eq(rule)
+  end
+
+  context "peer_with" do
+
+    it "should raise an error if the cidrs are overlapping" do
+      vpc_a = Terrafying::Components::VPC.create("a", "10.0.0.0/16")
+      vpc_b = Terrafying::Components::VPC.create("b", "10.0.0.0/20")
+
+      expect {
+        vpc_a.peer_with(vpc_b)
+      }.to raise_error(RuntimeError)
+    end
+
+    it "should create routes in both VPCs" do
+      vpc_a = Terrafying::Components::VPC.create("a", "10.0.0.0/16")
+      vpc_b = Terrafying::Components::VPC.create("b", "10.1.0.0/16")
+
+      original_route_count = vpc_a.output_with_children["resource"]["aws_route"].count
+
+      vpc_a.peer_with(vpc_b)
+
+      num_new_routes = vpc_a.output_with_children["resource"]["aws_route"].count - original_route_count
+
+      expect(num_new_routes).to eq(vpc_a.subnets.count * vpc_a.azs.count + vpc_b.subnets.count * vpc_b.azs.count)
+    end
+
+    it "should allow users to limit the subnets that are peered" do
+      vpc_a = Terrafying::Components::VPC.create("a", "10.0.0.0/16")
+      vpc_b = Terrafying::Components::VPC.create("b", "10.1.0.0/16")
+
+      original_route_count = vpc_a.output_with_children["resource"]["aws_route"].count
+
+      vpc_a.peer_with(vpc_b, { our_subnets: [vpc_a.subnets[:public]],
+                               their_subnets: [vpc_b.subnets[:public]] })
+
+      num_new_routes = vpc_a.output_with_children["resource"]["aws_route"].count - original_route_count
+
+      expect(num_new_routes).to eq(1 * vpc_a.azs.count + 1 * vpc_b.azs.count)
```

Comment:
why `1*`?

Reply:
It was an attempt at clarity against the other related tests to show there was only 1 expected subnet for each. I might try and make it clearer